dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentityattributevalue/TrackerTrigramIndexingJob.java

```diff
@@ -0,0 +1,145 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.trackedentityattributevalue;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.hisp.dhis.commons.collection.CollectionUtils;
+import org.hisp.dhis.scheduling.Job;
+import org.hisp.dhis.scheduling.JobConfiguration;
+import org.hisp.dhis.scheduling.JobProgress;
+import org.hisp.dhis.scheduling.JobType;
+import org.hisp.dhis.scheduling.parameters.TrackerTrigramIndexJobParameters;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.trackedentity.TrackedEntityAttributeService;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Ameen
+ */
+@Slf4j
+@Component( "trackerTrigramIndexingJob" )
+public class TrackerTrigramIndexingJob implements Job
+{
+    private final TrackedEntityAttributeService trackedEntityAttributeService;
+
+    public TrackerTrigramIndexingJob( TrackedEntityAttributeService trackedEntityAttributeService )
+    {
+        checkNotNull( trackedEntityAttributeService );
+        this.trackedEntityAttributeService = trackedEntityAttributeService;
+    }
+
+    // -------------------------------------------------------------------------
+    // Implementation
+    // -------------------------------------------------------------------------
+
+    @Override
+    public JobType getJobType()
+    {
+        return JobType.TRACKER_SEARCH_OPTIMIZATION;
+    }
+
+    @Override
+    public void execute( JobConfiguration jobConfiguration, JobProgress progress )
+    {
+        TrackerTrigramIndexJobParameters parameters = (TrackerTrigramIndexJobParameters) jobConfiguration
+            .getJobParameters();
+
+        log.info( "Starting Trigram Indexing Job. Attributes Provided to Index: {}", parameters.getAttributes() );
+        progress.startingProcess( "Starting Trigram indexing process" );
+
+        if ( !CollectionUtils.isEmpty( parameters.getAttributes() ) )
+        {
+            log.debug( "Fetching all indexable attributes from db" );
+            Set<TrackedEntityAttribute> indexableAttributes = trackedEntityAttributeService
+                .getAllTrigramIndexableTrackedEntityAttributes();
+
+            log.debug( "Found total {} indexable attributes", indexableAttributes.size() );
+            indexableAttributes = indexableAttributes.stream()
+                .filter( itea -> parameters.getAttributes().contains( itea.getUid() ) ).collect( Collectors.toSet() );
+
+            log.debug( "Number of Attributes provided in jobparameters that are indexable: {}",
```

Comment:
I would log all lower case in all the logs (except for maybe the SQL keywords you log). I would also stick to `job parameters` in all the logs. You use `jobparameters`, `jobParameters` (which might be typos 😄 )

Reply:
Good catch!